spec/02-integration/05-proxy/10-balancer/01-healthchecks_spec.lua

```diff
@@ -1056,17 +1089,23 @@ for _, strategy in helpers.each_strategy() do
               server1:start()
               server2:start()
 
-              oks, fails = bu.client_requests(bu.SLOTS * 2, api_host)
-              assert.same(bu.SLOTS * 2, oks)
-              assert.same(0, fails)
+              -- XXX: flaky
```

Comment:
I think this may be because the health check events have not been synchronized.

```suggestion
              helpers.wait_until(function()
               oks, fails, last_status = bu.client_requests(bu.SLOTS * 2, api_host)

               local ok, err = pcall(function ()
                assert.same(200, last_status)
                assert.same(bu.SLOTS * 2, oks)
                assert.same(0, fails)
               end)

               if not ok then
                 return false, err
               end

               return true
              end, 5)
```

It might make more sense to do so.